components/automate-ui/src/app/page-components/simple-line-graph/simple-line-graph.component.ts

```diff
@@ -0,0 +1,356 @@
+import {
+  Component, Input, OnChanges, ViewChild, ElementRef, SimpleChanges, OnInit
+} from '@angular/core';
+import * as d3 from 'd3';
+import {
+  DayPercentage
+} from 'app/entities/desktop/desktop.model';
+
+
+@Component({
+  selector: 'app-simple-line-graph',
+  templateUrl: './simple-line-graph.component.html',
+  styleUrls: ['./simple-line-graph.component.scss']
+})
+
+export class SimpleLineGraphComponent implements OnChanges, OnInit {
+
+  // labels too high
+  // disappearing render
+
+  constructor(
+    private chart: ElementRef
+  ) { }
+
+  @ViewChild('svg', { static: true }) svg: ElementRef;
+  @Input() data: DayPercentage[] = [];
+  @Input() width = 900;
+  @Input() height = 156; // we want a 116px height on the ticks, this minus margins
+  private locked: number = null;
+  private t = 1000; // transition speed
```

Comment:
Consider a more descriptive variable name. See my note on 'd' later for some elucidation on this point.